### DISCUSSION TITLE
Help with external subtitles for LG C9 (WebOS 4) or transcoding

### DISCUSSION THREAD

**Discussion by schafdog** [gerbera/gerbera#3322] [Q&A]
I can't get external subtitles (.srt) to work on a LG C9. Running gerbera 2.3-transcoding and importing has been set up with:
```
 <resources case-sensitive="no">
      <subtitle>
	<add-file name="%filename%.srt"/>
	<add-dir  name="subs" ext="srt"/>
	<add-dir  name="%filename%" ext="srt"/>
      </subtitle>
      <resource>
	<add-file name="%filename%.srt"/>
        <add-dir  name="subs" ext="srt"/>
        <add-dir  name="%filename%" ext="srt"/>
      </resource>
      <container location="" parentCount="2" minDepth="2"></container>
    </resources>
  ...
 extension-mimetype has
  <map from="srt" to="text/plain" />

mimetype-dlnatransfermode:
<map from="text/*" to="Background" />

```

I can confirm that subtitles are present in the details view of gerbera and can also see them with VLC. 

I haven't been able to confirm that the TV supports external subtitles, but haven't seen other people having reported issues. 
 
So LG C9 does not support this, I guess setting up transcoding for srt would be the solution, but haven't found an example configuration. The documentation describes that it is a possibility

**Comment by schafdog** [comment]
Manage to run it with debug:
```
<?xml version="1.0"?>
<item id="3461749" parentID="49451 8" restricted="1">
  <dc:title>A Movie</dc:title>
  <upnp:class>object.item.videoItem</upnp:class>
  <dc:created>2013-01-14</dc:created>
  <dc:date>2024-12-24T13:39:30+0000</dc:date>
  <upnp:playbackCount>108</upnp:playbackCount>
  <upnp:lastPlaybackTime>2024-12-25 T 13:51:11</upnp:lastPlaybackTime>
  <upnp:lastPlaybackPosition>0:00:00.000</upnp:lastPlaybackPosition>
  <upnp:albumArtURI xmlns:dlna="urn:schemas-dlna-org:metadata-1-0" dlna:profileID="JPEG_TN">http://10.0.0.16:49494/media/object_id/3461749/res_id/1/ext/file.jpg</upnp:albumArtURI>
  <sec:CaptionInfoEx protocolInfo="http-get:*:application/x-subrip:*" sec:type="srt">http://10.0.0.16:49494/media/object_id/3461749/res_id/2/type/srt/ext/file.srt</sec:CaptionInfoEx>
  <res size="2901310881" duration="3:22:06.280" bitrate="239258" sampleFrequency="48000" nrAudioChannels="2" resolution="1920x1080" protocolInfo="http-get:*:video/mp4:DLNA.ORG_PN=AVC_MP4_EU;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01700000000000000000000000000000" bitsPerSample="16" sec:acodec="aac" sec:vcodec="h264" orientation="1">http://10.0.0.16:49494/media/object_id/3461749/res_id/0/group/default/ext/file.mp4</res>
  <res resolution="160x90" protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00800000000000000000000000000000">http://10.0.0.16:49494/media/object_id/3461749/res_id/1/ext/file.jpg</res>
  <res protocolInfo="http-get:*:application/x-subrip:DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00d00000000000000000000000000000">http://10.0.0.16:49494/media/object_id/3461749/res_id/2/type/srt/ext/file.srt</res>
</item>
```

Wonder why the mime-type isn't "text/plain" for .srt

**Reply by KarlStraussberger** [reply]
Mime types for files are determined during import with https://docs.gerbera.io/en/latest/config-import.html#extension-mimetype

You can modify the mimetype for the device with https://docs.gerbera.io/en/latest/config-clients.html#client

If the file format itself is the problem, there is no way of converting the resource file with a transcoding rule. You can prepare multiple subtitle files and your device should pick it.

**Reply by KarlStraussberger** [reply]
If you add a new content type, you may also have to set https://docs.gerbera.io/en/latest/config-import.html#mimetype-dlnatransfermode

**Reply by schafdog** [reply]
Success! Not sure what did it. I had all configured, reimported but also upgraded to 2.4.1

**Comment by schafdog** [comment]
I can see that the protocol information changed from 
`protocolInfo="http-get:*:application/x-subrip:...`
to 
`protocolInfo="http-get:*:text/srt:..."`
